R/checks.R

```diff
@@ -533,3 +533,11 @@ check_gp_failure <- function(current, prev) {
   # return prev model
   prev
 }
+
+check_control <- function(control, name) {
+  if (!inherits(control, name)) {
+    rlang::abort(
```

Comment:
Since this is a user-facing error, I would set `call = NULL` in `rlang::abort` so that the user isn't shown:

```
Error in `check_control()` 
```

prepending the error.